src/context/mapStateSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { Map, Set } from 'immutable';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from './store';
+import { LayerType } from '../config/types';
+
+interface LayersSet extends Set<LayerType> {}
+interface MapState extends Map<string, LayersSet | Date> {}
```

Comment:
This *might* want to be `Map<string, Any>`? In general more specific types are better, but this state could end up growing into a pretty large union of types depending on what we add...